### PR TITLE
[SVLS-4999] Add Lambda tags to metrics sent via the API

### PR DESCRIPTION
**Pull request by DylanLovesCoffee**
### What does this PR do?

When the Extension is present, the Lambda library will use the DD API to send distribution metrics with timestamps instead of the Extension, since the Extension does not support historical distribution metrics. Metrics sent via the API currently do not get tagged with Lambda tags. 

This adds some essential tags like those added via `DD_TAGS` and the `function_arn`. With the `function_arn` tag available, custom metrics will be enriched with other tags collected by the metadata crawler if the AWS integration is enabled:
Screenshot 2024-07-02 at 3 55 16 PM

Also added a warning log for historical distribution metrics that are sent to the API with an unsupported timestamp (>=4hrs).

### Motivation

https://github.com/DataDog/datadog-lambda-js/issues/545 

### Testing Guidelines

Added tests and manually tested with local builds of the library.

### Types of Changes

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: datadog_lambda/metric.py [hunk 1]
```diff
@@ -61,6 +62,16 @@ def lambda_metric(metric_name, value, timestamp=None, tags=None, force_async=Fal
...
+        timestampCeiling = int(
+            (datetime.now() - timedelta(hours=4)).timestamp()
+        )  # 4 hours ago
+        if timestampCeiling > timestamp:
+            logger.warning(
+                "Timestamp %s is older than 4 hours, not submitting metric %s",
+                timestamp,
+                metric_name,
+            )
+            return
```
Comment (duncanista) [inline datadog_lambda/metric.py:74]: use snake case for `timestamp_ceiling`

Comment (DylanLovesCoffee) [reply datadog_lambda/metric.py:74]: 💀

**Review by duncanista — APPROVED** [review]
LGTM

**Pull request merged**